### ISSUE TITLE
Error using supergui (line 122)

### ISSUE THREAD

**Issue by Janinajj** [sccn/BCILAB#48]
Hi, 

I am using MATLAB R2017a and BCILAB version 1.1-beta (the only version I found) and when following the tutorial https://www.youtube.com/watch?v=Sup2SaJyFQE I get stuck when choosing the BCILAB paradigm. 
The following error appears:

Error using supergui (line 122)
supergui error: argument 'fig' must be numeric

Error in inputgui (line 161)
            [tmp tmp2 allobj] = supergui( 'fig', fig, 'minwidth', 200, 'geomhoriz', g.geometry, 'uilist', g.uilist, 'geomvert',
            [g.geomvert(:)' 1 1] );

Error in arg_guidialog (line 188)
[outs,dummy,okpressed] = inputgui('geometry',geometry, 'uilist',uilist,'helpcom',['env_doc ' helptopic],
'title',dialogtitle,'geomvert',geomvert); %#ok<ASGLU>

Error in gui_configapproach (line 38)
result = arg_guidialog(calibrate_func,'params',approach.parameters,'title','BCILAB: Configure approach','Invoke',false);

Error in gui_newapproach>gui_newapproach_OutputFcn (line 65)
        [varargout{1:2}] = gui_configapproach(varargout{1},true);

Error in gui_mainfcn (line 264)
        feval(gui_State.gui_OutputFcn, gui_hFigure, [], gui_Handles);

Error in gui_newapproach (line 28)
    gui_mainfcn(gui_State, varargin{:});
 
Error while evaluating Menu Callback.

Any idea how to solve this issue? 
Thanks

**Comment by dmedine** [comment]
I am also struggling with this and related errors. Basically the problem 
is that starting (I think) with Matlab 2014, Mathworks changed 
significantly how its graphics engine works. BCILab, however, is 
unmaintained and didn't keep up with the changes. Unfortunately this is 
true for a lot of toolboxes. Fortunately for EEGLab users, there is 
someone around to take care of keeping that code up to date with 
Mathworks. The work around (I think, please correct me if I am making a 
mistake) is to use an older version (<2014) of Matlab.

I'd be interested to know how many people are using BCILab on a regular 
basis and how they have been dealing with the fact that it is slowly 
self-deprecating.

**Comment by dmedine** [comment]
Haha! Actually, BCILAB is much more maintained than I had realized. You 
need a newer version. You will have to clone the git repository 
(https://github.com/sccn/BCILAB) and then switch to the devel branch. I 
just did this and it is working much better.

**Comment by Janinajj** [comment]
Awesome!!! That seems to work much better, thank you!

**Comment by Janinajj** [comment]
Still experiencing more issues though:
It can't find the target markers...

WARNING: the data set did not contain matching target markers.
WARNING: set_gettarget did not find any target information in this data set. See help of set_gettarget and set_targetmarkers for how data sets can be annotated with target information.

Computation failed; error trace:
 Some windows are entirely outside the data dimensions (0): [40 45;45 50]

Do you have any solutions to that?

**Comment by Janinajj** [comment]
Yes, he does, but it won't work.. have read about others having the same issue. 
Or does it work for you? Also saw the suggestion to download an older version of MATLAB.

**Comment by dmedine** [comment]
I have Matlab 2015b, I am running BCILab 1.5 devel and I had no problems 
whatsoever in following along with Christian's demo tutorial. Are you 
sure that the marker values you entered are free of typos? It should be: 
{{'S101', 'S102'}, {'S201', 'S202'}}

**Comment by roohiverma** [comment]
Hi,

I am new to MATLAB, BCILAB. 

I am on LUbuntu and using version R2018b version.

while running "env_compile_bcilab",  I am getting following error:

Creating dependency list /usr/local/MATLAB/R2018b/bin/BCILAB-1.1/build/dependency_list.m
Error during creation of the dependency list.
Error setting permissions for the dependency list.
Encountered a problem trying to copy the build/build-<arch>.reference file to the build.prj file.
Reason: Invalid file identifier. Use fopen to generate a valid file identifier.
Compiling BCILAB...
'deploytool' requires one of the following:
  MATLAB Compiler
  MATLAB Compiler SDK

Error in env_compile_bcilab (line 163)
deploytool('-build',env_translatepath('bcilab:/build.prj'));


If anyone knows, how to resolve, or any help would be really appreciated.

Thanks in advance !!!
